### DISCUSSION TITLE
TH Origin 20A

### DISCUSSION THREAD

**Discussion by titou4307** [arendst/Tasmota#16304] [Q&A] [answered]
Hello

I'm newbie with Tasmota ...

I have bought this product : 
https://sonoff.tech/product/diy-smart-switch/th-origin/

ESP is ESP32-D0WD-V3 (find in the link here and with esptool

I have made a backup of Oem firmware with esptool and after an erase of the ESP

I have use the online tool for import the firmware tasmota : 
https://tasmota.github.io/install/

Since this tool I have choose (because I don't know what to choice and of I have read it recommended to choose the first choice (tasmota english)

Does anyone could help me to put the good parameters ?
I have the sensor temperature RJ11 but it not appears...
I have the temperature of the ESP who appears ? for what ??
I have two link (may be for the 2 core ???)

So I am a little lost

Many thnaks for your helps

**Comment by sfromis** [comment]
For device operation you should apply the template for the device (see the how-to link to the left)
https://templates.blakadder.com/sonoff_THR320.html

The ESP32 temperature is generally a useless number, you can get rid of using `SetSensor127 0`

To get a value from the RJ11 temperature probe, you have to configure it. Based on other threads, my guess would be gpio25 as SI7021 in "Configure Template", after you installed/enabled the one above, but I know nothing.....

**Comment by titou4307** [comment, marked as answer]
HI
Many thanks for this response....
I will try this and so I return here

**Reply by titou4307** [reply]
Hello
I have try during this middlle of the days....
It seems more functionnal

You have explanation on the GPIO affectation ? There are GPIOx (01 to 17) and there are GPIO0x (014 to 035)

**Reply by sfromis** [reply]
You simply do not need to use the gpio info on the template page (format indeed seems a bit messed up, right now), just follow the how-to guide linked from the left column of the page. What you then get on the Configure Template page in the web UI is what counts.

(I've submitted a fix for the template which I hope will make the formatting work correctly)

**Reply by titou4307** [reply]
OK for this

Many thanks

**Reply by arendst** [reply]
![image](https://user-images.githubusercontent.com/11044339/186180870-5abf3375-67a6-4702-a0d9-953245aea122.png)

Select SI7021 for Sonoff TS01 or DS18x20 for DS18B20

**Reply by titou4307** [reply]
Perfect !!
Another response and by The Tasmota chief !!

**Reply by titou4307** [reply]
I have now another problem :

The Sonoff THR320 have on the main page this : 

![Sans titre 1](https://user-images.githubusercontent.com/56826797/186264643-4d8a1ebe-bb90-4e89-9867-18ad2eec5e93.png)

But only the relay appear on MQTT discover...
I have not the temperature..

**Reply by barbudor** [reply]
You are not clear
Do you mean the temp is not in the SENSOR MQTT message?
Or is that about Home Assistant?
You need 'setoption19 0' on Tasmota and HATASMOTA plug-in on HA
The 'old' 'setoption19 1' is now deprecated and no more maintained

**Reply by titou4307** [reply]
Sorry for my explanation

I'm using Gladys Assistant like home assistant
In gladys there broker who is Eclipse Mosquitto under container docker

Under Gladys there is a direct discover of integration Tasmota (protocol MQTT so it needeed the broker.....)

My Sonoff is discover but not with the two function (relay and sensor)
Or in the sonoff under tasmota I have the two function (relay and sensor)

So I think two options :
Opt 1 : tasmota Sonoff not provide the sensor information
Opt 2 : broker MQTT not find this....

does it this clear ?

**Reply by barbudor** [reply]
Not really much clearer
Are you using `SetOption19 1` ?
Does Gladys expect discovery messages using Home Assistant standard or Tasmota standard ?

**Reply by titou4307** [reply]
> Not really much clearer Are you using `SetOption19 1` 

Sorry I cannot give more
If I know the problem I will found the solution... : )

I will try your option

**Reply by barbudor** [reply]
please open a topic in the Gladys Assistant forum (the french one is fine) and point me to it, I will ping the developpers about it

**Comment by titou4307** [comment]
I have done this : 

https://community.gladysassistant.com/t/thr320-sonoff-tasmota-resolution-problemes/7495

When I read the list of SetOption may be the 2 could be the solution....
But before I will use (at the end of this day....after work) a MQTT Client for see if Sonoff give info of sensor into network MQTT

Thanks

**Reply by titou4307** [reply]
Hello
I have downloaded client MQTT Explorer :

I have find in this software the message here : 

{"sn":{"Time":"2022-08-24T17:59:31","DS18B20":{"Id":"3C6404577D30","Temperature":23.4},"ESP32":{"Temperature":46.7},"TempUnit":"C"},"ver":1}

So my idea is the sensor mount into MQTT but not to Gladys Assistant

Your idea ??

**Reply by barbudor** [reply]
You don't tell on which topic you get this so I assume it is on `tele/nameofyourdevice/SENSOR`
As you have the temperature on the web page , I had no doubt that tasmota is publishing the temperature to MQTT correctly
The point is understanding what Gladys needs to discover that.
Some automation software could use the discovery messages published by Tasmota on `homeassistant/` (deprecated) or `tasmota/` (new), some may discover only analysing the standard messages, some need manual configuration.
I have no idea how Gladys is working so that is why I asked you to publish a message on their forum. Hopefully someone with better knowledge, or one of Gladys developper, will be able to tell us what they need from Tasmota to properly discover the sensor

**Reply by sfromis** [reply]
The format with `{"sn":` would be the `tasmota/discovery/<mac>/sensors` retained topic, but that should indeed be accompanied by similar `tele/nameofyourdevice/SENSOR` messages. The discovery message is specifically meant to allow backend software to discover what features the Tasmota device has, of course only if it "knows of" and supports that format. My expectation would be that adapting to the Tasmota-specific discovery format would be "less common" for non-HA backend platforms.

**Reply by barbudor** [reply]
Looking at the code of Gladys, it doesn't look like it is using any discovery. Just parsing the message.
What I just saw is that it only support standard FullTopic %prefix%/%topic%
And I feel it is not parsing the SENSOR message

**Comment by titou4307** [comment]
I make correction 

On MQTT Explorer  :

Under **tasmota/discovery/782184BC276C**

{"sn":{"Time":"2022-08-24T17:59:31","DS18B20":{"Id":"3C6404577D30","Temperature":23.4},"ESP32":{"Temperature":46.7},"TempUnit":"C"},"ver":1}

Under  **tele/tasmota_BC276C/SENSOR**

{"Time":"2022-08-24T19:09:29","DS18B20":{"Id":"3C6404577D30","Temperature":23.5},"ESP32":{"Temperature":46.1},"TempUnit":"C"}

**Reply by sfromis** [reply]
The last message indeed indicates that Tasmota is doing what it is supposed to. In Gladys, you need to have some place where it can parse the format published by Tasmota. That may not be an "oven-ready" feature, but I'd not be surprised if it has some sort of more generic parsing allowing you to pick what fields you want to use how. Or maybe you can write a bit of parsing code picking what you want from the JSON.

**Reply by barbudor** [reply]
Gladys parsing of tasmota message is here : https://github.com/GladysAssistant/Gladys/blob/master/server/services/tasmota/lib/mqtt/tasmota.mqtt.handleMessage.js
It's not using the discovery but I feel that it is looking for a `StatusSNS` key in the SENSOR message

**Reply by barbudor** [reply]
@titou4307 Please follow here : https://github.com/GladysAssistant/Gladys/issues/1590
This is a Gladys problem, not Tasmota

**Reply by sfromis** [reply]
For context, if it is looking for `StatusSNS`, that should not be in a message on topic `.../SENSOR` but `STATUS10` (or `STATUS8`).

**Reply by titou4307** [reply]
Ok I will try to ask to the developper of Gladys....

I have sent him a message with the topic on Gladys community and the 2 topics here
I hope he can take a little time for confirm and maybe upgrade this assistant

Many thanks for the time spended 

titou4307 (from France)

**Reply by titou4307** [reply]
Just for confirm I have another container with Home Assistant
I have add the MQTT broker of Gladys

Just for the fun I have add another sensor DS18x20 at my Sonoff Tasmota (possible after reading doc of tasmota...)

Here the result : 

![Sans titre 2](https://user-images.githubusercontent.com/56826797/186530366-dbfed0ae-885d-4eae-ad7e-cad4e01c8c7d.png)

It confirm the diagnostic this is Gladys bug
